File: app/src/test/java/com/dominikgaller/alpinebooking/booking/core/domain/tourbooking/TourDateTest.java
package com.dominikgaller.alpinebooking.booking.core.domain.tourbooking;

import org.junit.jupiter.api.Test;

import java.time.Instant;
import java.time.temporal.ChronoUnit;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatNullPointerException;

class TourDateTest {

    private static final Instant NOW = Instant.parse("2026-03-03T12:00:00Z");

    @Test
    void isInFutureReturnsTrueForTomorrow() {
        final TourDate tomorrow = new TourDate(
                NOW.plus(1, ChronoUnit.DAYS).atZone(java.time.ZoneOffset.UTC).toLocalDate());
        assertThat(tomorrow.isInFuture(NOW)).isTrue();
    }

    @Test
    void isInFutureReturnsFalseForYesterday() {
        final TourDate yesterday = new TourDate(
                NOW.minus(1, ChronoUnit.DAYS).atZone(java.time.ZoneOffset.UTC).toLocalDate());
        assertThat(yesterday.isInFuture(NOW)).isFalse();
    }

    @Test
    void isInFutureReturnsFalseForToday() {
        final TourDate today = new TourDate(
                NOW.atZone(java.time.ZoneOffset.UTC).toLocalDate());
        assertThat(today.isInFuture(NOW)).isFalse();
    }

    @Test
    void nullDateThrows() {
        assertThatNullPointerException().isThrownBy(() -> new TourDate(null));
    }

    @Test
    void nullNowThrows() {
        final TourDate date = new TourDate(java.time.LocalDate.of(2030, 1, 1));
        assertThatNullPointerException().isThrownBy(() -> date.isInFuture(null));
    }
}

File: src/test/kotlin/com/example/contractmanagement/booking/core/domain/tourbooking/TourDateTest.kt
package com.example.contractmanagement.booking.core.domain.tourbooking

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.time.Instant
import java.time.LocalDate

class TourDateTest {
    private val now: Instant = Instant.parse("2026-06-01T10:00:00Z")

    @Test
    fun isInFuture_isTrue_forATomorrowDate() {
        assertThat(TourDate(LocalDate.parse("2026-06-02")).isInFuture(now)).isTrue()
    }

    @Test
    fun isInFuture_isFalse_forTheSameDay() {
        assertThat(TourDate(LocalDate.parse("2026-06-01")).isInFuture(now)).isFalse()
    }

    @Test
    fun isInFuture_isFalse_forAPastDate() {
        assertThat(TourDate(LocalDate.parse("2026-05-31")).isInFuture(now)).isFalse()
    }
}
